lsm.go

```diff
@@ -255,11 +255,9 @@ func (l *LSM) findNode(node *Node) *Node {
 }
 
 func (l *LSM) Compact() error {
-	if l.compacting.CompareAndSwap(false, true) {
-		return l.compact()
+	for !l.compacting.CompareAndSwap(false, true) {
```

Comment:
Why not use the compaction pool sweep-style approach we had before?

Reply:
The compactor pool was implemented because we were compacting many granules in parallel which could cause go routine explosion. Here we only have the single routine at a time.

That said, we may absolutely need a compactor pool here but at this time I don't have enough signaling about how this behaves in practice to know so a compactor pool was a premature optimization.